src/utils/__tests__/internalLinks-test.js

```diff
@@ -363,3 +364,78 @@ describe('getMessageIdFromLink', () => {
     ).toBe(1);
   });
 });
+
+describe('getLinkToMessage', () => {
+  test('should return a stream link if message link copied from stream', () => {
+    const realm = 'https://zulip.com';
+    const message = eg.streamMessage({
+      stream_id: 12345,
+      display_recipient: 'stream',
+      subject: 'exampleTopic',
+    });
+
+    const result = getLinkToMessage(realm, message);
+    const expectedResult = `https://zulip.com/#narrow/stream/12345-stream/topic/exampleTopic/near/${message.id}`;
+
+    expect(result).toEqual(expectedResult);
+  });
+
+  test('should return a group link if message link copied from pm with >= 3 people', () => {
+    const realm = 'https://zulip.com';
+
+    const message = eg.pmMessage({
+      display_recipient: [{ id: 11111 }, { id: 22222 }, { id: 33333 }],
+    });
+
+    const result = getLinkToMessage(realm, message);
+    const expectedResult = `https://zulip.com/#narrow/pm-with/11111,22222,33333-group/near/${message.id}`;
+
+    expect(result).toEqual(expectedResult);
+  });
+
+  test('should return pm link for message link copied from pm with < 3 people', () => {
+    const realm = 'https://zulip.com';
+    const message = eg.pmMessage({
+      display_recipient: [{ id: 11111 }, { id: 22222 }],
+    });
+
+    const result = getLinkToMessage(realm, message);
+    const expectedResult = `https://zulip.com/#narrow/pm-with/11111,22222-pm/near/${message.id}`;
+
+    expect(result).toEqual(expectedResult);
+  });
+
+  test('properly encodes topics with tricky characters', () => {
+    const { realm } = eg.selfAccount;
+    const message = eg.streamMessage({
+      stream_id: 123,
+      display_recipient: 'general',
+      subject: '.%%2E/#\ud83d\udc4b',
+    });
+    expect(getLinkToMessage(realm, message)).toEqual(
+      `${realm}/#narrow/stream/123-general/topic/.2E.25.252E.2F.23.F0.9F.91.8B/near/${message.id}`,
```

Comment:
You may also want to add a non-ASCII Unicode character in the BMP (i.e., one not represented by a pair of surrogates in UTF-16).